dotCMS/src/main/java/com/dotcms/cache/VanityUrlCacheImpl.java

```diff
@@ -115,58 +119,68 @@ public void remove(final Contentlet vanityURL) {
             if (Host.SYSTEM_HOST.equals(siteId)) {
 
                 //Get all the sites related to these cached VanityURLs
-                Set<String> sites = getSiteIds();
+                final Set<String> sites = getSiteIds();
                 if (null != sites) {
                     for (String site : sites) {
 
-                        if (!Host.SYSTEM_HOST.equals(site)) {
-
-                            final List<CachedVanityUrl> siteCachedVanityUrl = this
-                                    .getCachedVanityUrls(VanityUrlUtil
-                                            .sanitizeSecondCacheKey(site,
-                                                    vanityURL.getLanguageId()));
-
-                            if (null != siteCachedVanityUrl) {
-                                Boolean deleteForHost = Boolean.FALSE;
-
-                                for (CachedVanityUrl toRemove : siteCachedVanityUrl) {
-
-                                    //We care only about 404 records
-                                    if (VanityUrlAPI.CACHE_404_VANITY_URL
-                                            .equals(toRemove.getVanityUrlId())) {
-
-                                        //Remove from the cache this 404 record
-                                        this.remove(VanityUrlUtil
-                                                .sanitizeKey(toRemove.getSiteId(),
-                                                        toRemove.getUrl(),
-                                                        toRemove.getLanguageId()));
-
-                                        //We found a 404 record, we need to remove this site from the cache
-                                        deleteForHost = Boolean.TRUE;
-                                        break;
-                                    }
-                                }
-
-                                if (deleteForHost) {
-                                    cleanBySecondaryRegion(VanityUrlUtil
-                                            .sanitizeSecondCacheKey(site,
-                                                    vanityURL.getLanguageId()));
-                                }
-                            }
-                        }
+                        this.removeSite(vanityURL, site);
                     }
                 }
             }
 
-            this.remove(VanityUrlUtil.sanitizeKey(vanityURL));
+            this.remove(new CacheVanityKey(
+                                vanityURL.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
+                                vanityURL.getLanguageId(),
+                                vanityURL.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
+                            ).toString());
 
-        } catch (DotDataException | DotRuntimeException | DotSecurityException e) {
+
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)